specs/ScrollingControls/ScrollPresenter.md

```diff
@@ -0,0 +1,2462 @@
+the current position of the content and default composition animation which is a Vector3KeyFrameAnimation about to be launched by default.
+The correlationId is the one provided earlier in ScrollControllerScrollToRequestedEventArgs.CorrelationId or
+in ScrollControllerScrollByRequestedEventArgs.CorrelationId.
+The scroll controller can then return: null or an unchanged _defaultAnimation_ to use that default animation, a modified version of _defaultAnimation_, or a brand-new custom animation.
+
+## IScrollController.NotifyScrollCompleted
+
+```cs
+void NotifyScrollCompleted(Int32 correlationId)
+```
+
+This method is invoked by the `ScrollPresenter` to indicate that a scrolling operation initiated through a
+ScrollToRequested, ScrollByRequested or AddScrollVelocityRequested event has completed.
+It provides the operation's correlationId which was exposed earlier in
+ScrollControllerScrollToRequestedEventArgs.CorrelationId, ScrollControllerScrollByRequestedEventArgs.CorrelationId or
+ScrollControllerAddScrollVelocityRequestedEventArgs.CorrelationId. |
+
+## IScrollController.ScrollToRequested  event
+
+```cs
+event Windows.Foundation.TypedEventHandler<IScrollController, ScrollControllerScrollToRequestedEventArgs> ScrollToRequested
+```
+The scroll controller can request a scroll to a particular offset by raising its ScrollToRequested event.
+The `ScrollPresenter` handles the event and provides a correlationId in ScrollControllerScrollToRequestedEventArgs.CorrelationId for
+the operation that is about to start.
+The scroll controller can for example make such a request when a thumb is dragged to a new position.
+
+## IScrollController.ScrollByRequested event
+
+```cs
+Windows.Foundation.TypedEventHandler<IScrollController, ScrollControllerScrollByRequestedEventArgs> ScrollByRequested
+```
+
+The scroll controller can request a scroll by a particular offset delta by raising its ScrollByRequested event.
+The `ScrollPresenter` handles the event and provides a correlationId in ScrollControllerScrollByRequestedEventArgs.CorrelationId for
+the operation that is about to start.
+The scroll controller can for example make such a request when a button is clicked to move a thumb by an increment.
+
+## IScrollController.AddScrollVelocityRequested 
+
+```cs
+event Windows.Foundation.TypedEventHandler<IScrollController, ScrollControllerAddScrollVelocityRequestedEventArgs> AddScrollVelocityRequested
+```
+
+The scroll controller can request a scroll by adding velocity to the `ScrollPresenter` content through the AddScrollVelocityRequested event.
+
+## IScrollController.InteractionRequested event
+
+```cs
+event Windows.Foundation.TypedEventHandler<IScrollController, ScrollControllerInteractionRequestedEventArgs> InteractionRequested
+```
+
+This event can be raised by the scroll controller when the user attempts to initiate a UI-thread-independent pan,
+with touch or a pen, using the UIElement returned by the InteractionElement property.
```

Comment:
It's only for the off thread case? So the controller raises the event off thread?